v2/pkg/archive/archive.go

```diff
@@ -160,11 +176,24 @@ func (o MirrorArchive) addBlobsDiff(collectedBlobs, historyBlobs map[string]stri
 	return blobsInDiff, nil
 }
 
-func (o MirrorArchive) addFile(pathToFile string, pathInTar string) error {
+func (o *MirrorArchive) addFile(pathToFile string, pathInTar string) error {
 	fi, err := os.Stat(pathToFile)
 	if err != nil {
 		return err
 	}
+
+	// when a file is already bigger than the maxArchiveSize, it will not fit any chunk
+	// therefore we should stop
+	if fi.Size() > o.maxArchiveSize {
+		return fmt.Errorf("maxArchiveSize %dG is too small compared to sizes of files that need to be included in the archive.\n%s: %dG\n Aborting archive generation", o.maxArchiveSize/segMultiplier, fi.Name(), fi.Size()/segMultiplier)
```

Comment:
@sherine-k - in this scenario, (just a suggestion) would it not be a better user experience to actually log a warning message and try work out what the max size is (some simple algorithm)  ?
This way the user doesn't need to guess and also at the very last step of mirroring have a failure. 
Also what about the use of --continue-on-error flag ?

Reply:
PermissiveAdder is now the default adder interface for MirrorArchive.
StrictAdder is accessible via flag --strict-archive